packages/common/src/graphql/entities.ts

```diff
@@ -1,14 +1,126 @@
 // Copyright 2020-2021 OnFinality Limited authors & contributors
 // SPDX-License-Identifier: Apache-2.0
 
-import {GraphQLSchema, GraphQLObjectType, isObjectType} from 'graphql';
+import {
+  GraphQLSchema,
+  GraphQLObjectType,
+  isObjectType,
+  isNonNullType,
+  getNullableType,
+  isListType,
+  assertListType, getDirectiveValues, GraphQLField
+} from 'graphql';
 import {DirectiveName} from './constant';
 
+import {
+  GraphQLModelsRelations,
+  GraphQLModelsType,
+  GraphQLEntityField,
+  FieldScalar,
+  GraphQLRelationsType
+} from './types';
+
 export function getAllEntities(schema: GraphQLSchema): GraphQLObjectType[] {
   return Object.entries(schema.getTypeMap())
     .filter(([, node]) => node.astNode?.directives?.find(({name: {value}}) => value === DirectiveName.Entity))
     .map(([, node]) => node)
     .filter(isObjectType);
 }
 
-// TODO: GraphQLNamedType -> EntitySchema
+export function getAllEntitiesRelations(schema: GraphQLSchema): GraphQLModelsRelations{
+  const modelRelations = {models:[],relations:[]} as GraphQLModelsRelations;
+  const entityNameSet = [];
+  const reverseLookupRelations = [];
+
+  const derivedFrom = schema.getDirective('derivedFrom');
+  const entities = Object.entries(schema.getTypeMap())
+      .filter(([, node]) => node.astNode?.directives?.find(({name: {value}}) => value === DirectiveName.Entity))
+      .map(([, node]) => node)
+      .filter(isObjectType);
+
+  for (const entity of entities){
+    entityNameSet.push(entity.name);
+  }
+
+  for (const entity of entities){
+    const newModel: GraphQLModelsType=  {
+      name: entity.name,
+      fields : []
+    }
+    const fields = entity.getFields();
+    for (const k in fields){
+      if (Object.prototype.hasOwnProperty.call(fields, k)) {
+        const typeString = extractType(fields[k].type).toString();
+        //If is a basic scalar type
+        if (Object.values(FieldScalar).includes(typeString)) {
+          newModel.fields.push(packEntityField(typeString, fields[k],'basic'));
+        }
+        // If is a foreign key
+        else if (entityNameSet.includes(typeString) && fields[k].astNode.directives.length ===0){
```

Comment:
`fields[k].astNode.directives.length ===0` is not the only scenario that `@deriveFrom` isn't present, you better improve the condition branches otherwise it is likely you will have a lot of duplicated codes